pallets/staking/src/lib.rs

```diff
@@ -12,86 +12,149 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+//! # Liquid staking pallet
+//!
+//! This pallet manages the NPoS operations for relay chain asset.
+
 #![cfg_attr(not(feature = "std"), no_std)]
-#![allow(clippy::unused_unit)]
-#![allow(clippy::collapsible_if)]
 
-use frame_support::transactional;
-use frame_support::{pallet_prelude::*, PalletId};
+use frame_support::{pallet_prelude::*, transactional, PalletId};
 use frame_system::pallet_prelude::*;
+use sp_runtime::{traits::AccountIdConversion, FixedPointNumber, RuntimeDebug};
+
 use orml_traits::{MultiCurrency, MultiCurrencyExtended};
-use primitives::{Amount, Balance, CurrencyId};
-use sp_runtime::{traits::AccountIdConversion, RuntimeDebug};
-use sp_std::convert::TryInto;
-use sp_std::vec::Vec;
 
-pub use module::*;
+pub use pallet::*;
+use primitives::{Amount, Balance, CurrencyId, Rate};
 
-mod staking;
+#[cfg(test)]
+mod mock;
+
+#[cfg(test)]
+mod tests;
 
 /// Container for pending balance information
 #[derive(Encode, Decode, Eq, PartialEq, Copy, Clone, RuntimeDebug, Default)]
-pub struct PendingBalance<Moment> {
-    pub balance: Balance,
-    pub timestamp: Moment,
+pub struct UnstakeInfo<BlockNumber> {
+    pub amount: Balance,
+    pub block_number: BlockNumber,
 }
 
 #[frame_support::pallet]
-pub mod module {
+pub mod pallet {
     use super::*;
 
+    #[pallet::pallet]
+    #[pallet::generate_store(pub(super) trait Store)]
+    pub struct Pallet<T>(_);
+
     #[pallet::config]
-    pub trait Config: frame_system::Config + pallet_timestamp::Config {
+    pub trait Config: frame_system::Config {
+        /// The overarching event type.
         type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
 
-        /// Currency type for deposit/withdraw collateral assets to/from loans
-        /// module
+        /// Currency type used for staking and liquid assets
         type Currency: MultiCurrencyExtended<
             Self::AccountId,
             CurrencyId = CurrencyId,
             Balance = Balance,
             Amount = Amount,
         >;
 
-        /// The loan's module id, keep all collaterals of CDPs.
+        /// Currency used for staking
+        #[pallet::constant]
+        type StakingCurrency: Get<CurrencyId>;
+
+        /// Currency used for liquid voucher
```

Comment:
I think voucher isn't the same thing with staking mecanism as voucher means `money replacement`, people can spend it.

However, staking just ensure that people can get money back, so I think it's not the same thing